.github/workflows/ci-update-workflow.yml

```diff
@@ -0,0 +1,62 @@
+name: ci-update-workflow
+
+on:
+  schedule:
+  # Run every hour.
+  - cron:  '0 * * * *'
+
+jobs:
+  update-go-versions:
+    runs-on: ubuntu-latest
+    steps:
+    - id: 'auth'
+      uses: 'google-github-actions/auth@v1'
+      with:
+        credentials_json: '${{ secrets.GCP_CREDENTIALS }}'
+
+    - name: 'Set up Cloud SDK'
+      uses: 'google-github-actions/setup-gcloud@v1'
+      with:
+        version: '423.0.0'
+
+    - name: Get available Go versions
+      run: |
+        # Get all supported Go versions and pass the space-separated versions
+        # string to the next step
+        versions=$(gcloud functions runtimes list --format="value(NAME)" --project=gae-runtimes-private --region=us-west1 | grep ^go | tr '\n' ' ' | xargs)
+        echo "::set-output name=versions::$versions"
+      id: get_versions
+
+    - name: Checkout functions-framework-go repo
+      uses: actions/checkout@v2
+
+    - name: Update Go versions
+      run: |
+        readarray -d ' ' -t versions <<< "${{ steps.get_versions.outputs.versions }}"
+        
+        versions_str=""
+        for version in "${versions[@]}"; do
+          major=${version:2:1}
+          minor=${version:3:2}
+          versions_str+="'$major.$minor.x', "
+        done
+        # Remove the last comma and space
+        versions_str=${versions_str%, *}
+        # Only replace the first occurance
+        sed -i "0,/^        go: \[.*/s//        go: \[$versions_str\]/" .github/workflows/unit.yml
+        sed -i "0,/^        go: \[.*/s//        go: \[$versions_str\]/" .github/workflows/lint.yml
+  
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v4
+      with:
+        token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
```

Comment:
We might want to avoid using PATs. 
Github provides a special token {{secrets.GITHUB_TOKEN}} which expires after a job runs.
More info in: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret
Lets verify if it is a good fit here